### PR TITLE
Bootstrap to Flipper

### PR DESCRIPTION
**Pull request by saraycp**
Uses Flipper instead of Feature to handle Bootstrap feature.

### CODE REVIEW

File: src/api/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -89,6 +89,13 @@ def self.validate_action(opt)
 
   protected
 
+  def feature_enabled?(feature)
+    return if Flipper.enabled?(feature.to_sym, User.possibly_nobody) || Flipper.enabled?(feature.to_sym)
```
Comment (DavidKang) [inline src/api/app/controllers/application_controller.rb]: Why `User.possibly_nobody`? IMO we should user `User.session` that is not taking into account `_nobody_` user.

File: src/api/app/controllers/webui/webui_controller.rb [hunk 2]
```diff
@@ -378,4 +385,8 @@ def set_influxdb_additional_tags
   def set_webui2_views
     prepend_view_path('app/views/webui2')
   end
+
+  def enabled_feature?(feature)
```
Comment (hennevogel) [inline src/api/app/controllers/webui/webui_controller.rb]: `feature_enabled` & `enabled_feature` confuse me..

File: src/api/spec/lib/feature_toggle/flipper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    end
+
+    context 'with user both in beta and rollout' do
+      let(:user) { create(:confirmed_user, :in_beta, login: 'Tom') }
+
+      before do
+        User.session = user
+      end
+
+      it { expect(subject).to be_truthy }
```
Comment (DavidKang) [inline src/api/spec/lib/feature_toggle/flipper_spec.rb]: This shouldn't be falsey? :thinking:

File: src/api/app/controllers/application_controller.rb [hunk 4]
```diff
@@ -89,6 +89,13 @@ def self.validate_action(opt)
...
+    render file: Rails.root.join('public/404'), status: :not_found, layout: false
+  end
+
+  # TODO: remove when all the migration from Feature to Flipper is finished.
+  # It'll be replaced by feature_enabled?
   def feature_active?(feature)
     return if Feature.active?(feature)
     render file: Rails.root.join('public/404'), status: :not_found, layout: false
```
Comment (vpereira) [inline src/api/app/controllers/application_controller.rb]: why it cannot return simply boolean without render?

**Comment by saraycp** [comment]
[Move bootstrap from the old feature gem to flipper](https://trello.com/c/nhbtzglL/1184-move-bootstrap-from-the-old-feature-gem-to-flipper)

File: src/api/app/controllers/application_controller.rb (hunk 4 above)
Comment (DavidKang) [reply src/api/app/controllers/application_controller.rb]: @vpereira, We use this method as a `before_action` into different controllers like `image_template` to make them not accessible.

Comment (saraycp) [reply src/api/app/controllers/application_controller.rb]: Exactly, useful when someone goes directly to a URL that only exists if the feature is active. But if the user is not allowed to access to the feature, he/she will see the 404.

File: src/api/app/controllers/application_controller.rb (hunk 1 above)
Comment (saraycp) [reply src/api/app/controllers/application_controller.rb]: done!

**Comment by saraycp** [comment]
@hennevogel please have another look.

**Pull request merged**